lms/security.py

```diff
@@ -120,19 +146,25 @@ def identity(self, request):
 
         return Identity("", [])
 
-    def permits(self, request, context, permission):
-        return _permits(self, request, context, permission)
+    def permits(self, request, _context, permission):
+        return _permits(self.identity(request), permission)
 
 
-def _permits(policy, request, _context, permission):
-    identity = policy.identity(request)
+def _permits(identity, permission):
     if identity and permission in identity.permissions:
         return Allowed("allowed")
 
     return Denied("denied")
 
 
-def _get_lti_user(request) -> LTIUser:
+def _get_lti_user_from_lti_launch_params(request) -> LTIUser:
+    schema = LTI11AuthSchema(request).lti_user
+    if "id_token" in request.params:
+        schema = LTI13AuthSchema(request).lti_user
+    return schema()
```

Comment:
Would early-return make this function a little simpler? I think the below works the same? But no need to keep track of a local variable `schema`.

```python
def _get_lti_user_from_lti_launch_params(request) -> LTIUser:
    if "id_token" in request.params:
        return LTI13AuthSchema(request).lti_user()

    return LTI11AuthSchema(request).lti_user()
```